### DISCUSSION TITLE
Switch expression precedence?

### DISCUSSION THREAD

**Discussion by AlgorithmsAreCool** [dotnet/roslyn#53490] [Q&A] [answered]
Hello, I wrote some code today using a switch expression and got a confusing result.

I expected each of these functions to return 8 when given an input of 2.

But instead the GetPaddingExpr returns 2, which appears to be incorrect.

After some fiddling I discovered that the switch expression was "binding" to the literal 8 instead of the entire `value % 8` expression.

This is somewhat unexpected to me, is this the intended behavior?

```csharp
using System;

Console.WriteLine($"GetPaddingExpr -> {GetPaddingExpr(2)}");
Console.WriteLine($"GetPaddingExprFixed -> {GetPaddingExprFixed(2)}");
Console.WriteLine($"GetPaddingStatement -> {GetPaddingStatement(2)}");

static int GetPaddingExpr(int value)
{
    return
        value % 8 switch {
            0 => value,
            _ => ((value / 8) + 1) * 8 
        };
}

static int GetPaddingExprFixed(int value)
{
    return
        (value % 8) switch {
            0 => value,
            _ => ((value / 8) + 1) * 8 
        };
}

static int GetPaddingStatement(int value)
{
    switch(value % 8)
    {
        case 0: return value;
        default: return((value / 8) + 1) * 8;
    }
}

```

**Comment by jasonmalinowski** [comment]
Hmm, @CyrusNajmabadi, @sharwell, this might be a good candidate for us suggesting parenthesis in this case? Heuristic would be that a user probably doesn't want to do a switch expression on a numerical constant.

**Reply by CyrusNajmabadi** [reply]
i think this would fall under teh standard parentehses precedence suggestions.  It's just that we likely need to update that code for handlign switch expressions :)

**Comment by CyrusNajmabadi** [comment, marked as answer]
> This is somewhat unexpected to me, is this the intended behavior?

Yes.  This is the intended behavior.  `switch` has higher precedence than `%`, so your code is effectively parsed as if it ws:

```c#
        value % (8 switch {
            0 => value,
            _ => ((value / 8) + 1) * 8 
        });
```
